corehq/apps/locations/permissions.py

```diff
@@ -20,34 +21,57 @@ def is_locations_admin(view_fn):
     return locations_access_required(domain_admin_required(view_fn))
 
 
-def user_can_edit_location(user, location, project):
-    """
-    Expects SQLLocation
-    """
-    if not project.location_restriction_for_users:
+@quickcache(['user._id', 'project.name'])
```

Comment:
Couldn't find a good way to cache these locations inside tastypie, so using quickcache instead